### PR TITLE
feat: create a child client on every search

### PR DESCRIPTION
**Pull request by ccalbaugh**
This will allow us to pass headers and things like requestTimeout on a per search basis.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  This PR has failing tests. Please alleviate the errors and commit them

<details>
  <summary>runSearch should wrap queries in constant_score if sort._score is not present</summary>
  <code>Cannot read property 'args' of undefined</code>
</details>
<details>
  <summary>runSearch should not wrap queries in constant_score if no query is given</summary>
  <code>Cannot read property 'args' of undefined</code>
</details>
<details>
  <summary>runSearch should not wrap queries in constant_score if sort._score is present</summary>
  <code>Cannot read property 'args' of undefined</code>
</details>
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 7d67168b675d4f9b20f72eb19ef9cbbddbf7398c


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 952e60be19cabf7226da054980d4396c711ec6d9
</p>

**Comment by ccalbaugh** [comment]
Converting to draft until the tests are passing again.

**Comment by ccalbaugh** [comment]
Ready for review!

**Review by rma4ok — APPROVED** [review]
LGTM

**Comment by ccalbaugh** [comment]
After merging the conflicts the tests are failing, so changing to wip until those are passing.

File: src/index.js [hunk 1]
```diff
@@ -47,12 +47,16 @@ let ElasticsearchProvider = (config = { request: {} }) => {
           }
 
       let client = config.getClient()
+      let child = client.child({
+        headers: requestOptions.headers,
+        requestTimeout: requestOptions.requestTimeout,
```
Comment (daedalus28) [inline src/index.js:52]: does this mean that we shouldn't be passing `requestOptions` to `search` anymore?

File: src/index.js [hunk 2]
```diff
@@ -47,12 +47,16 @@ let ElasticsearchProvider = (config = { request: {} }) => {
           }
 
       let client = config.getClient()
+      let child = client.child({
```
Comment (daedalus28) [inline src/index.js:50]: nit: Assuming client isn't referenced anywhere else, you could reduce the diff size with something like

```js
let child = config.getClient().child({ ....
```

**Review by daedalus28 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (ccalbaugh) [reply src/index.js:52]: Well this encompasses all the needs that we are using for requestOptions, but you could always pass other things to search within those options that we just aren't using.

File: src/index.js (hunk 2 above)
Comment (ccalbaugh) [reply src/index.js:50]: Sure I'll update that 👍

File: src/index.js (hunk 1 above)
Comment (daedalus28) [reply src/index.js:51]: got it - so should we take headers/requestTimeout off before passing it along?

Comment (daedalus28) [reply src/index.js:51]: given that we were passing it along before, it's probably fine. Merging :)

**Pull request merged**